pages/deployment/interactivity.mdx

```diff
@@ -2,81 +2,227 @@
+
+This flow is bi-directional. If a user interacts with the dashboard and changes a variable (e.g., by using a filter), you can listen to these changes in your app (see below).
+
+## Listen to Variable Changes
+
+You can listen to variable changes in your Embeddable by adding an event listener for the `onVariablesChange` event. This event is fired each time a variable inside your Embeddable updates (e.g., user selects a new country). The event detail contains an array of objects with the variable name and its new value.
+
+Here is an example of how to listen to variable changes in your Embeddable:
+
+```tsx
+const securityToken = await getSecurityTokenFromYourBackend();
+function handleVariableChange(e) {
+  console.log('Variables changed:', e.detail);
+  // e.detail is an array of { variableName, newValue }
+  // eg:
+  // detail: [
+  //   { variableName: 'Country', newValue: 'France' },
+  //   { variableName: 'Date range', newValue: { from: '2024-01-01', to: '2024-12-31' } }
+  // ]
+}
+
+// Embeddable React component defined as per example below
+<Embeddable
+  token={securityToken}
+  onVariablesChange={handleVariableChange}
+/>
+```
+
+## Handle Component Errors
+
+You can listen to errors that occur in your Embeddable by adding an event listener for the `onEmbeddableError` event. This event is fired each time an error occurs in your Embeddable (e.g., network error, invalid token). The event's `detail` property contains an object with the error message and error detail.
+
+Here is an example of how to listen to errors in your Embeddable:
+
+```tsx
+const securityToken = await getSecurityTokenFromYourBackend();
+function handleEmbeddableError(e) {
+  console.error('Embeddable error:', e.detail);
+  // e.detail is an object with errorMessage and errorDetail
+  // eg:
+  // detail: {
+  //   errorMessage: 'Network error',
+  //   errorDetail: 'Failed to fetch data from API'
+  // }
+}
+
+// Embeddable React component defined as per example below
+<Embeddable
+  token={securityToken}
+  onEmbeddableError={handleEmbeddableError}
+/>
+```
+
+You can use this to log errors, show error messages to users, or take other actions in response to errors. You could also use this to send the error details to your back-end for further processing, such as alerting or logging.
+
+## Know When Components are Loaded
+
+You can listen to when your Embeddable has finished loading its components by adding an event listener for the `onComponentsLoad` event. This event is fired each time the Embeddable has finished loading all its components (e.g., charts, tables). This can be useful for showing a loading spinner while the Embeddable is loading, and then hiding it once the components are ready.
```

Comment:
@cwbuecheler I think it's worth making it clear that this will be fired once the components have loaded and rendered (usually in a loading state). It does not wait for any loadData requests to return.